firebase-sessions/src/main/kotlin/com/google/firebase/sessions/FirebaseSessions.kt

```diff
@@ -26,146 +25,53 @@ import com.google.firebase.ktx.Firebase
 import com.google.firebase.ktx.app
 import com.google.firebase.sessions.api.FirebaseSessionsDependencies
 import com.google.firebase.sessions.api.SessionSubscriber
-import com.google.firebase.sessions.settings.SessionsSettings
+import com.google.firebase.sessions.leader.SessionMaintainerLeader
 import kotlinx.coroutines.CoroutineDispatcher
 
-/** The [FirebaseSessions] API provides methods to register a [SessionSubscriber]. */
+/**
+ * The [FirebaseSessions] API provides methods to register a [SessionSubscriber] and delegates to a
+ * [SessionMaintainer].
+ */
 class FirebaseSessions
 internal constructor(
-  private val firebaseApp: FirebaseApp,
+  firebaseApp: FirebaseApp,
   firebaseInstallations: FirebaseInstallationsApi,
   backgroundDispatcher: CoroutineDispatcher,
   blockingDispatcher: CoroutineDispatcher,
   transportFactoryProvider: Provider<TransportFactory>,
 ) {
-  private val applicationInfo = SessionEvents.getApplicationInfo(firebaseApp)
-  private val sessionSettings =
-    SessionsSettings(
-      firebaseApp.applicationContext,
-      blockingDispatcher,
-      backgroundDispatcher,
-      firebaseInstallations,
-      applicationInfo,
-    )
-  private val timeProvider: TimeProvider = Time()
-  private val sessionGenerator: SessionGenerator
-  private val eventGDTLogger = EventGDTLogger(transportFactoryProvider)
-  private val sessionCoordinator = SessionCoordinator(firebaseInstallations, eventGDTLogger)
-
-  init {
-    sessionGenerator = SessionGenerator(collectEvents = shouldCollectEvents(), timeProvider)
 
-    val sessionInitiateListener =
-      object : SessionInitiateListener {
-        // Avoid making a public function in FirebaseSessions for onInitiateSession.
-        override suspend fun onInitiateSession(sessionDetails: SessionDetails) {
-          initiateSessionStart(sessionDetails)
-        }
-      }
+  private val sessionMaintainer: SessionMaintainer
+  private val tag = "FirebaseSessions"
 
-    val sessionInitiator =
-      SessionInitiator(
-        timeProvider,
+  init {
+    // TODO(rothbutter): create a different maintainer based on the characteristics of this process
```

Comment:
Curiousity question: How is the current maintainer leader different from the TODO maintainer and what characteristics does this depend on?

Reply:
I assure you that your curiosity will be satisfied in the next CL :)

Suffice to say that it will be significantly different from the leader.